first iteration

dranikpg opened the pull request:
- two small issues with hset commands
- we have very little support for server-management commands, most of them are stubs / don't react to help either, I removed the subcommands

dranikpg commented on docs/command-reference/server-management/client-list.md:
````diff
@@ -34,58 +31,12 @@ Here is the meaning of the fields:
-```
-
-The file descriptor events can be:
+* `phase`: TODO
````
How should I describe this?

"whether the connection is currently processing a command"

The states are: process and readsock.

romange replied:
todo is fine. maybe `state of the connection as captured by "client list" command`

dranikpg commented on docs/command-reference/server-management/client.md:
```diff
@@ -12,4 +12,4 @@ description: A container for client connection commands
 
-To see the list of available commands you can call `CLIENT HELP`.
\ No newline at end of file
+<!-- To see the list of available commands you can call `CLIENT HELP`. -->
```
We'd better add the help commands before 1.0

dranikpg commented on docs/command-reference/server-management/command.md:
```diff
@@ -69,65 +56,16 @@ Examples:
-The only commands that require using `COMMAND GETKEYS` are `SORT` and `MIGRATE` for clients that parse keys' specifications.
-
-For more information, please refer to the [key specifications page][tr].
+// TODO: should we describe our internal flags?
```
should we describe things like reverse-mapping and no-autojournal? I guess they shouldn't be user-facing

dranikpg replied:
I left no description and instead filed an issue https://github.com/dragonflydb/dragonfly/issues/861

dranikpg commented on docs/command-reference/server-management/hello.md:
```diff
@@ -6,24 +6,13 @@ description: Handshake with Redis
-In Redis 6 connections start in RESP2 mode, so clients implementing RESP2 do
-not need to updated or changed. There are no short term plans to drop support for
-RESP2, although future version may default to RESP3.
+At the moment, the RESP3 protocol is not supported.
```
HELLO currently makes little sense for switching protocols, maybe just remvoe the argument?

dranikpg commented on docs/command-reference/server-management/info.md:
```diff
@@ -11,240 +11,106 @@ description: Get information and statistics about the server
-
-# Modules
-
-# Errorstats
```
In Df errors stats is not part of default

dranikpg commented on docs/command-reference/server-management/info.md:
```diff
@@ -261,13 +127,12 @@ INFO
 
-Here is the description of fields for Redis >= 2.4.
-
+<!-- TODO
```
😢

dranikpg commented on docs/command-reference/server-management/role.md:
````diff
@@ -92,4 +92,5 @@ The sentinel output is composed of the following parts:
 ROLE
 ```
 
+// TODO: What about the slave note?
 **A note about the word slave used in this man page**: Starting with Redis 5, if not for backward compatibility, the Redis project no longer uses the word slave. Unfortunately in this command the word slave is part of the protocol, so we'll be able to remove such occurrences only when this API will be naturally deprecated.
````
Should we keep such a note that we use the term slave only for backward comp?

romange replied:
We should remove this disclaimer.

dranikpg commented on docs/command-reference/server-management/save.md:
```diff
@@ -6,20 +6,15 @@ description: Synchronously save the dataset to disk
-of an RDB file.
+The `SAVE` commands performs a save of the dataset producing a
+_point in time_ snapshot of all the data inside the Dragonfly instance, in the form
+of a collection of DFS files.
```
collection of dfs files sounds not good

romange replied:
"DFS" abbreviation must be explained. collection -> group -> set ?

dranikpg replied:
Lets call it 'set of DFS files' and make it point to the serialization page

dranikpg commented on docs/command-reference/server-management/info.md:
```diff
@@ -610,52 +249,17 @@ The **commandstats** section provides statistics based on the command type,
-**A note about the word slave used in this man page**: Starting with Redis 5, if not for backward compatibility, the Redis project no longer uses the word slave. Unfortunately in this command the word slave is part of the protocol, so we'll be able to remove such occurrences only when this API will be naturally deprecated.
-
-**Modules generated sections**: Starting with Redis 6, modules can inject their info into the `INFO` command, these are excluded by default even when the `all` argument is provided (it will include a list of loaded modules but not their generated info fields). To get these you must use either the `modules` argument or `everything`.,
+// TODO: slace question
```
.

dranikpg commented on docs/command-reference/server-management/info.md:
```diff
@@ -261,118 +127,45 @@ INFO
-*   `lazyfreed_objects`: The number of objects that have been lazy freed.
+// TODO: Custom flags
 
+// TODO: How do make users check fragmentation ratio? `MEMORY MALLOC-STATS`?
+<!--
 Ideally, the `used_memory_rss` value should be only slightly higher than
 `used_memory`.
 When rss >> used, a large difference may mean there is (external) memory fragmentation, which can be evaluated by checking
```
we should add this to the docs

romange reviewed (commented):
why did you delete "auth" command?

romange reviewed (commented):
we also support "memory help"

romange commented on docs/command-reference/server-management/role.md:
```diff
@@ -10,7 +10,7 @@ description: Return the role of the instance in the context of replication
 **Time complexity:** O(1)
 
-Provide information on the role of a Redis instance in the context of replication, by returning if the instance is currently a `master`, `slave`, or `sentinel`. The command also returns additional information about the state of the replication (if the role is master or slave) or the list of monitored master names (if the role is sentinel).
+Provide information on the role of a Dragonfly instance in the context of replication, by returning if the instance is currently a `master`, `slave`, or `sentinel`. The command also returns additional information about the state of the replication (if the role is master or slave) or the list of monitored master names (if the role is sentinel).
```
we never return `sentinel`

romange reviewed (commented):
Left a few commands.

dranikpg commented:
I added auth (by accident), cut out the ACL section
I added MEMORY HELP back but its not very helpful as we support one command and the other parts contains a "not supported" note
